### DISCUSSION TITLE
Web: Rescript for web/playground

### DISCUSSION THREAD

**Discussion by palerdot** [PRQL/prql#4863] [Ideas]
I would like to propose [Rescript](https://rescript-lang.org/) for web/playground codebase. Rescript would be a good choice because
- excellent type system (it is Ocaml underneath)
- first-class react support (since React is used by web/playground codebases)
- fast compiler (builds, error messages are almost instant)
- less cognitive overload for rust developers to switch between rust code and web/playground codebase (this is subjective).
- current npm/package.json based workflow will seamlessly work
- Rescript can also be gradually adopted (though adopting it in full would offer all the benefits)

I was looking into fixing an issue in playground (https://github.com/PRQL/prql/issues/4720). I was less confident of making changes primarily because of untyped codebase (and also because of quaint class based react code). If the codebase was typed (like typescript), I would be more confident in making changes.

I think Rescript would be a good choice for prql's web codebase. I'm not sure if Rescript was evaluated before and also would be happy to contribute to a POC (playground codebase written in Rescript for example) if needed.

**Comment by max-sixty** [comment]
I am very aligned with Rescript's goals! And a big fan of OCaml (as you can maybe tell by the syntatic sparsity of PRQL)

But because the playground is not the core part of PRQL, it needs to be open to more informal contributors, which means we need to use a well-known language. It's not the place within PRQL to make big technical bets.

I think typescript would be very reasonable, but I would counsel against something like Rescript. The one exception is if someone is really keen to build it out in a way that's better than what exists, is prepared to own it, and can credibly signal that (for example, they've already built it, or have built something similar and stuck with it for a while).

Is that reasonable? If others have thoughts please add!

**Reply by vanillajonathan** [reply]
> I think typescript would be very reasonable, but I would counsel against something like Rescript.

I agree.
